### DISCUSSION TITLE
ente admin fails

### DISCUSSION THREAD

**Discussion by roughnecks** [ente-io/ente#1417] [Q&A] [answered]
Hello,
I think to have a working self-hosted instance, since my photos are uploaded on an external S3 Storage.
Now I wanted to remove the storage limits with "ente admin" command, but it's failing.

```
./ente admin list-users

Assuming "myemailaddrees" as the Admin
------------
Error: status 500 with err:
Usage:
  ente admin list-users [flags]

Flags:
  -a, --admin-user string   The email of the admin user.
  -h, --help                help for list-users
  ```

and in docker console I see:

```
museum-1  | ERRO[0027]handler.go:46 Error Request failed                                error= --- at /etc/ente/pkg/api/admin.go:104 (AdminHandler.GetUsers) ---
museum-1  |  --- at /etc/ente/pkg/repo/user.go:123 (UserRepository.GetAll) ---
museum-1  | Caused by: email decryption failed
```

I have previously added my account with `./ente account add` and I can see it with `./ente account list`
that's where I got the id to be an admin.

**Comment by mnvr** [comment]
The error is coming from this function in [repo/user.go](https://github.com/ente-io/ente/blob/main/server/pkg/repo/user.go#L121). Very likely, this means that there is some mismatch in the server's encryption keys (the default configuration and documentation for these is in [local.yaml](https://github.com/ente-io/ente/blob/main/server/configurations/local.yaml#L151)).

So you should try to fix your configuration, because a server key mismatch will cause other issues. That said, meanwhile if you just want to see the user ID, you can get them from the SQL db (or print them in the museum code above where the err occurred).

**Comment by roughnecks** [comment]
I know about the encryption keys and I generated new keys with the command specified in the commented section, put them in `museum.yaml`, then started my containers and volumes over (docker compose down --volumes). So what does it mean? That I should use default keys and cannot change them?
I want to remove the storage limits.. Already have the id to be admin.

Thanks

**Reply by mnvr** [reply]
You can change them (and you should change them, the sample values are not safe to use). 

But maybe you created your admin account before changing them, so now it is unable to decrypt the data that was encrypted with the older key. Is that a possibility?

If you're sure the keys were the same throughout, then let me know, I can take a look to see if there is maybe some issue in the command for generating new keys.

**Reply by roughnecks** [reply]
I'm quite sure to have taken docker down before applying new keys and I also made a brand new account with a different email.
The only thing I'm not sure about is if I have to generate the keys inside the container or I can use a cloned ente repository like I did.
I'm not building docker images, but using the pre-built ones.

**Reply by mnvr** [reply]
That explains it though - while you created a brand new account, there still are old accounts that were using the older key (taking docker down does not delete the volumes where the DB is stored). So now when we're trying to "list-all", the decryption of the emails for the pre-key-change accounts is failing.

Easiest would be to start with a clean slate for the DB, making sure that the volumes are also getting deleted when the cluster is stopped 
```
docker compose down --volumes
```
(note: this will delete all the data in the DB that you have so far. make sure that's what you want to do in your case).

**Reply by mnvr** [reply]
BTW, just for completeness' sake I'll mention that there is another way to increase the storage quota - by directly modifying the DB. e.g. see this section in the guide someone has written for a similar scenario as yours - https://help.ente.io/self-hosting/guides/external-s3#_4-create-an-account-and-increase-storage-quota

**Reply by roughnecks** [reply, marked as answer]
Yeah, that was my intention and I actually did run `docker compose down --volumes` but I have postgres in my host, not as a docker container, so I should delete that db too, I guess.

**Reply by roughnecks** [reply]
I have deleted psql db and started over, everything is working as expected now.

Thanks a bunch and sorry if I'm marking my own comment as answer :)

**Reply by mnvr** [reply]
No worries, glad this got sorted! Cheers
